### PR TITLE
Travis CI coverage for Emacs 27

### PR DESCRIPTION
**Pull request by peterbecich**
Converts Travis CI to Nix.
Needs `kubectl` installed with Nix to fix a test: https://travis-ci.com/github/peterbecich/kubernetes-el/jobs/376627133#L6520

### CODE REVIEW

**Comment by chrisbarrett** [comment]
yesssss 👍 

4cq59p

**Comment by peterbecich** [comment]
@chrisbarrett It looks like this has caught an actual error: 
```
kubernetes-loading-container.el:18:18:Error: `seq-contains' is an obsolete function (as of 27.1); use `seq-contains-p' instead.
```
https://travis-ci.org/github/chrisbarrett/kubernetes-el/jobs/721921919#L6486

File: kubernetes-loading-container.el [hunk 1]
```diff
@@ -15,7 +15,7 @@
   (cond
    ((null vector)
     on-loading)
-   ((and vector (seq-contains-p vector elem))
+   ((and vector (if (fboundp 'seq-contains-p) (seq-contains-p vector elem) (seq-contains vector elem)))
```
Comment (peterbecich) [inline kubernetes-loading-container.el:18]: Trying this: https://github.com/watofundefined/doom-emacs-private/blob/6a66041a711e2b22f0b3034bcaa1b4fc419243e8/modules/wat-utils.el#L21-L23

**Comment by chrisbarrett** [comment]
@peterbecich Looking good! Let me know when you're done and I'll merge. :D

File: kubernetes-loading-container.el [hunk 2]
```diff
@@ -15,7 +15,11 @@
   (cond
    ((null vector)
     on-loading)
-   ((and vector (if (fboundp 'seq-contains-p) (seq-contains-p vector elem) (seq-contains vector elem)))
+   ((and vector
+         (if
+             (fboundp 'seq-contains-p)
+             (seq-contains-p vector elem)
+           (with-no-warnings (seq-contains vector elem))))
```
Comment (peterbecich) [inline kubernetes-loading-container.el:22]: copied this https://emacs.stackexchange.com/a/9802/6682

**Comment by peterbecich** [comment]
@chrisbarrett , would you merge this as-is?  Emacs 25.2-3, 26.1-3 and 27.1 succeed: https://travis-ci.org/github/chrisbarrett/kubernetes-el/builds/725818024

I have disabled the Emacs 25.1 job.  It fails: https://travis-ci.org/github/chrisbarrett/kubernetes-el/jobs/725816066#L7264
I think either Travis has cached a broken artifact for the 25.1 job, or the artifact is not available for 25.1:
```
  - Installing [ 8/10] evil (latest)... downloadingDependency evil failed to install: Failed to verify signature: "undo-tree-0.7.5.el.sig"
```
The first possibility might be ruled out by deleting the repository's Travis CI caches and re-implementing the 25.1 job: https://github.com/chrisbarrett/kubernetes-el/blob/5a1c807fa2645b42df68300405c6c24fd17af588/.travis.yml#L9-L13

I am inquiring whether Cask might be able to throw a more informative error message here: https://github.com/cask/cask/issues/488

**Comment by chrisbarrett** [comment]
Excellent. Thanks very much for shaving this yak!

**Comment by peterbecich** [comment]
@chrisbarrett You're welcome, but I see some of the jobs failed.  Would you try deleting the Travis caches and restarting the build, please?

**Pull request merged**